app/components/secure/WalletSecureComponent.tsx

```diff
@@ -2,88 +2,64 @@ import * as React from 'react';
+    const { Theme, AppConfig } = useAppConfig();
     const safeArea = useSafeAreaInsets();
-    const reboot = useReboot();
+    let disableEncryption =
+        props.deviceEncryption === 'none'
+        || props.deviceEncryption === 'device-biometrics'
+        || props.deviceEncryption === 'device-passcode'
+        || Platform.OS === 'android' && Platform.Version < 30 // Encryption doesn't work well on older androids
 
     // Action
     const [loading, setLoading] = React.useState(false);
-    const onClick = React.useCallback((bypassEncryption?: boolean) => {
+    const onClick = React.useCallback(() => {
         (async () => {
             setLoading(true);
             try {
 
-                // Encrypted token
-                let secretKeyEnc: Buffer;
-
                 // Generate New Key
-                try {
-                    let disableEncryption = !!(props.deviceEncryption === 'none' || props.deviceEncryption === 'device-biometrics' || props.deviceEncryption === 'device-passcode' || bypassEncryption);
-                    if (Platform.OS === 'android' && Platform.Version < 30) {
-                        disableEncryption = true; // Encryption doesn't work well on older androids
-                    }
-                    secretKeyEnc = await generateNewKeyAndEncrypt(disableEncryption, Buffer.from(props.mnemonics));
-                } catch (e) {
-                    // Ignore
-                    console.warn(e);
-                    return;
-                }
+                let secretKeyEnc = await generateNewKeyAndEncrypt(
+                    false,
+                    Buffer.from(props.mnemonics)
+                );
 
                 // Resolve key
                 const key = await mnemonicToWalletKey(props.mnemonics.split(' '));
 
-                // Resolve utility key
-                const utilityKey = await deriveUtilityKey(props.mnemonics.split(' '));
-
                 // Resolve contract
                 const contract = await contractFromPublicKey(key.publicKey);
 
-                // Persist state
-                const state = getAppState();
-                setAppState({
-                    addresses: [
-                        ...state.addresses,
-                        {
-                            address: contract.address,
-                            publicKey: key.publicKey,
-                            secretKeyEnc,
-                            utilityKey,
-                        }
-                    ],
-                    selected: state.addresses.length
-                }, AppConfig.isTestnet);
-
-                // Persist secured flag
-                if (props.import) {
-                    markAddressSecured(contract.address, AppConfig.isTestnet);
-                }
+                // Save to storage and default state to Use biometrics
+                storeBiometricsEncKey(contract.address.toFriendly({ testOnly: AppConfig.isTestnet }), secretKeyEnc);
```

Comment:
maybe it can be in generateNewKeyAndEncrypt?

Reply:
we do this to set the tag in storage before we have Engine running where we could set it with state in atom like we do in BiometricsSetupFragment